src/main/java/io/github/fbiville/nestedrelationquerysdn6/HypercubeNode.java

```diff
@@ -3,13 +3,27 @@
 import org.springframework.data.neo4j.core.schema.GeneratedValue;
 import org.springframework.data.neo4j.core.schema.Id;
 import org.springframework.data.neo4j.core.schema.Node;
+import org.springframework.data.neo4j.core.schema.Property;
 
 @Node(labels = "Hypercube")
 public abstract class HypercubeNode {
 
-    @Id
-    @GeneratedValue
-    private Long id;
+	@Id
+	@GeneratedValue
+	private Long internalID;
+
+	@Property("ID")
+	private Long id;
```

Comment:
is this guaranteed to be unique for each entity? If so, that would be a good candidate for `@Id`.

Reply:
We were planning to have a progressive ID starting from one to each entity. We may use the internalId for cross reference with other domains but ID as property just an incremental value.

Reply:
Be wary that the internal ID is not guaranteed to be stable over time (this is just an offset in the underlying store files and these files can be defragmented over time). It's usually not a good candidate for an external ID (SDN 6 provides a UUID provider if you want a technical, stable ID): https://docs.spring.io/spring-data/neo4j/docs/current/reference/html/#mapping.id-handling